mycore-base/src/main/java/org/mycore/resource/MCRResourcePath.java

```diff
@@ -126,36 +151,85 @@ private static MCRResourcePath createResourcePath(String path) {
+    }
+
+    /**
+     * Creates a {@link MCRResourcePath} that represents the given path as a web resource path,
+     * i.e. a resource path that implicitly starts with <code>/META-INF/resources/</code>.
+     * <p>
+     * Returns an empty {@link Optional} in case of a missing or empty path or in case
+     * of a path targeting a directory.
+     * <p>
+     * Throws a {@link IllegalPathException} in case of a path starting with <code>/META-INF</code>
+     * or <code>/WEB-INF</code> or in case of a path targeting a class.
      */
     public static Optional<MCRResourcePath> ofWebPath(String path) {
-        return cleanAndSafePath(path).map(MCRResourcePath::createWebResourcePath);
+        return Optional.ofNullable(cleanAndSafePath(path, false)).map(MCRResourcePath::createWebResourcePath);
+    }
+
+    /**
+     * Creates a {@link MCRResourcePath} that represents the given path as a web resource path,
+     * i.e. a resource path that implicitly starts with <code>/META-INF/resources/</code>.
+     * <p>
+     * Throws a {@link IllegalPathException} in case of a missing or empty path or in case
+     * of a path targeting a directory or in case of a path starting with <code>/META-INF</code>
+     * or <code>/WEB-INF</code> path or in case of a path targeting a class.
+     */
+    public static MCRResourcePath ofWebPathOrThrow(String path) {
+        return createWebResourcePath(cleanAndSafePath(path, true));
     }
 
     private static WebResourcePath createWebResourcePath(String path) {
-        if (path.startsWith("/META-INF/") || path.startsWith("/WEB-INF/") || path.endsWith(".class")) {
-            throw new MCRResourceException("Path points to undeliverable web resource: " + path);
+        if (path.startsWith("/META-INF/")) {
+            throw IllegalPathException.Code.PATH_POINTS_TO_META_INF_RESOURCE.createException(path);
+        }
+        if (path.startsWith("/WEB-INF/")) {
+            throw IllegalPathException.Code.PATH_POINTS_TO_WEB_INF_RESOURCE.createException(path);
+        }
+        if (path.endsWith(".class")) {
+            throw IllegalPathException.Code.PATH_POINTS_TO_CLASS_RESOURCE.createException(path);
         }
         LOGGER.debug("Creating web resource path {}", path);
         return new WebResourcePath(path);
     }
 
-    private static Optional<String> cleanAndSafePath(String path) {
-        if (path == null || path.isEmpty() || path.endsWith("/")) {
-            return Optional.empty();
-        } else if (!path.startsWith("/")) {
-            return Optional.of(safePath("/" + path));
-        } else {
-            return Optional.of(safePath(path));
+    private static String cleanAndSafePath(String path, boolean orThrow) {
+        if (path == null || path.isEmpty()) {
+            if (orThrow) {
+                throw IllegalPathException.Code.MISSING_OR_EMPTY_PATH.createException(path);
+            }
+            return null;
```

Comment:
Mabe an explit else {} would make it clear, that the result is based on the `orThrow` parameter? - Or is this a PMD validation? - check other occurences as well